cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -197,6 +198,13 @@ public PushAppResponse pushApp(ApplicationReference appRef, String commitMessage
     RepositoryConfig repoConfig = getRepositoryMeta(appRef.getParent()).getConfig();
     ApplicationDetail appDetail = appLifecycleService.getLatestAppDetail(appRef, false);
 
+    ApplicationId appId = appRef.app(appDetail.getAppVersion());
+    accessEnforcer.enforce(appId, authenticationContext.getPrincipal(), StandardPermission.GET);
```

Comment:
This requires reading the app but also pushing to the repo, seems like it would involve a edit permission on the repo. We currently don't have a repository entity, but should we?

Reply:
As offline discussed, we need new permission of Repository Pull, Push and List. Will add the code change in a separate PR. This PR should be adding necessary checks on the current role and permission we have